core/services/llo/mercurytransmitter/transmitter.go

```diff
@@ -183,6 +194,31 @@ func (mt *transmitter) Start(ctx context.Context) (err error) {
 					go s.runDeleteQueueLoop(mt.stopCh, mt.wg)
 					go s.runQueueLoop(mt.stopCh, mt.wg, donIDStr)
 				}
+				mt.collectors = append(mt.collectors, prometheus.NewGaugeFunc(
```

Comment:
Why a `GaugeFunc` here? 

What about a util func that wraps recording a new gauge value? You could then set the labels directly on the instantiated instrument. Would that save drilling the Registerer down the stack?

Reply:
@samsondav 's comments from offline, which I agree with:

> I could have done it that way but GaugeFunc felt like the right tool for this problem, and it should have been straightforward if we were actually using prometheus properly (app-scoped registry). It should be dependency-injected like the logger